wurst/systems/spawns/BushSelection.wurst

```diff
@@ -1,23 +1,43 @@
-package BushSelection
-import HashMap
-import RegisterEvents
-import InventoryUtils
-import Classes
-import PlayerExtensions
-import ID
-
-let MAX_DIST = 300
-let MAX_DIST_SQ = compiletime(MAX_DIST * MAX_DIST)
-
-let map = new HashMap<int, int>()
-..put(UNIT_THIEFS_BUSH, UNIT_THIEF)
-..put(UNIT_SCOUTS_BUSH, UNIT_SCOUT)
-
-init
-    registerPlayerUnitEvent(EVENT_PLAYER_UNIT_SELECTED) ->
-        let bush = GetTriggerUnit()
-        let hero = GetTriggerPlayer().getTroll()
-        let id = bush.getTypeId()
-
-        if map.has(id) and map.get(id) == getTrollBaseClass(hero) and hero.getPos().distanceToSq(bush.getPos()) < MAX_DIST_SQ
-            bush.transferItems(hero)
+package BushSelection
+import HashMap
+import RegisterEvents
+import InventoryUtils
+import Classes
+import PlayerExtensions
+import ID
+
+let MAX_DIST = 300
+let MAX_DIST_SQ = compiletime(MAX_DIST * MAX_DIST)
+
+let map = new HashMap<int, int>()
+    ..put(UNIT_THIEFS_BUSH, UNIT_THIEF)
+    ..put(UNIT_SCOUTS_BUSH, UNIT_SCOUT)
+
+function transferItems(unit u, player p)
+    let id = map.get(u.getTypeId())
+
+    // Exit if the selection unit is not a bush.
+    if id == 0
+        return
+
+    // Enumerate all units that may be close enough to receive items.
+    GroupEnumUnitsInRange(
+        ENUM_GROUP,
+        u.getPos().x,
+        u.getPos().y,
+        MAX_DIST.toReal(),
+        Filter(-> true)
+    )
+
```

Comment:
Im not sure if you consider this cleaner but
```
let units = CreateGroup()..
    enumUnitsInRange(pos, radius, null)
```

Reply:
I wasn't aware of that method. I'll use it when writing up `findNearestUnit`.